classes/Ajax.php

```diff
@@ -754,4 +759,48 @@ public function delete_annoucement() {
 
 		wp_send_json_error( array( 'message' => __( 'Announcement delete failed', 'tutor' ) ) );
 	}
+
+	/**
+	 * Get youtube video duration.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function ajax_youtube_video_duration() {
+		tutor_utils()->check_nonce();
+
+		$video_id = Input::post( 'video_id' );
+		if ( empty( $video_id ) ) {
+			$this->json_response( __( 'Video ID is required', 'tutor' ), null, HttpHelper::STATUS_BAD_REQUEST );
+		}
+
+		$has_access = User::is_admin() || User::is_instructor();
```

Comment:
Instead of this
```
		$has_access = User::is_admin() || User::is_instructor();
		if ( ! $has_access ) {
			$this->json_response( tutor_utils()->error_message(), null, HttpHelper::STATUS_BAD_REQUEST );
		}
```
We can write this
`tutor_utils()->check_current_user_capability('publish_tutor_courses')`;

Reply:
We can't use this because, instructor need to be approved to take this action.